libraries/AP_GPS/AP_GPS_GSOF.cpp

```diff
@@ -36,28 +40,23 @@
 
 extern const AP_HAL::HAL& hal;
 
-#define gsof_DEBUGGING 0
-
-#if gsof_DEBUGGING
-# define Debug(fmt, args ...)                  \
-do {                                            \
-    hal.console->printf("%s:%d: " fmt "\n",     \
-                        __FUNCTION__, __LINE__, \
-                        ## args);               \
-    hal.scheduler->delay(1);                    \
-} while(0)
-#else
-# define Debug(fmt, args ...)
-#endif
-
 AP_GPS_GSOF::AP_GPS_GSOF(AP_GPS &_gps,
                          AP_GPS::Params &_params,
                          AP_GPS::GPS_State &_state,
                          AP_HAL::UARTDriver *_port) :
     AP_GPS_Backend(_gps, _params, _state, _port)
 {
+    gsofmsgreq.set(AP_GSOF::POS_TIME);
+    gsofmsgreq.set(AP_GSOF::POS);
+    gsofmsgreq.set(AP_GSOF::VEL);
+    gsofmsgreq.set(AP_GSOF::DOP);
+    gsofmsgreq.set(AP_GSOF::POS_SIGMA);
+
     // https://receiverhelp.trimble.com/oem-gnss/index.html#GSOFmessages_Overview.html?TocPath=Output%2520Messages%257CGSOF%2520Messages%257COverview%257C_____0
-    static_assert(ARRAY_SIZE(gsofmsgreq) <= 10, "The maximum number of outputs allowed with GSOF is 10.");
```

Comment:
prefer compile time assert if possible, happy to have new constructor for AP_Bitmask that takes const array

Reply:
Done!